docs/messenger/messenger.rst

```diff
@@ -0,0 +1,122 @@
+
+Transports
+----------
+
+A transport is a connection to an endpoint that can receive and serve messages (understand work payload). For example, a synchronous transport has no endpoint and instead, the handler is called directly. In comparison, the RabbitMQ transport (AMQP) uses a RabbitMQ server as an endpoint and will send and fetch these messages from remote server. Doctrine transport uses a database.
+
+For more information, refer to the `Symfony documentation <https://symfony.com/doc/current/messenger.html#transports>`_.
+
+Failed Transport
+^^^^^^^^^^^^^^^^
+
+Failed transport in Symfony Messenger is a feature that helps with managing failed messages. A failed message is a message that could not be handled by a message handler due to an exception or error.
+
+Symfony Messenger automatically catches exceptions thrown by handlers. Depending on the transport configuration, the message can either be retried or sent to the "failure transport".
+
+This failure transport serves as a storage for these failed messages and allows you to view, debug, and retry the processing of these messages manually.
+
+
+Routing
+-------
+
+Every message needs to have a transport associated with it. The synchronous transport is a valid selection. By default, PageHit and EmailHit messages are routed to the **hit** transport, which is configured as synchronous.
+
+.. code-block:: php
+
+    MauticMessengerTransports::HIT    => [
+        'dsn'        => 'sync://',
+        'serializer' => 'messenger.transport.jms_serializer',   // Smaller payload
+    ],
+
+.. _default-configuration:
+
+The default configuration is as follows
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: php
+
+    'routing' => [
+        \Symfony\Component\Mailer\Messenger\SendEmailMessage::class => MauticMessengerTransports::EMAIL,
+        \Mautic\MessengerBundle\Message\PageHitNotification::class  => MauticMessengerTransports::HIT,
+        \Mautic\MessengerBundle\Message\EmailHitNotification::class => MauticMessengerTransports::HIT,
+    ],
+
+For more information on routing, refer to the `Symfony documentation <https://symfony.com/doc/current/messenger.html#routing-messages-to-a-transport>`_.
+
+
+Consuming Messages
+==================
+
+To consume or process messages from a transport, Symfony Messenger provides a console command ``messenger:consume``. You can specify the transport(s) as an argument, or it will consume messages from all configured transports by default.
+
+Here's an example of consuming messages from a transport:
+
+.. code-block:: bash
+
+    php bin/console messenger:consume your_transport
+
+Replace ``your_transport`` with the name of your transport.
+
+To consume messages from all transports, you would use:
+
+.. code-block:: bash
+
+    php bin/console messenger:consume
+
+The command will keep running and consume messages as they come until it's manually stopped.
```

Comment:
I'm getting this message when I try this:
```

www-data@28de97d2dce5:~/html$ bin/console messenger:consume

                                                                                                                        
 Which transports/receivers do you want to consume?                                                                     
                                                                                                                        

Choose which receivers you want to consume messages from in order of priority.
Hint: to consume from multiple, use a list of their names, e.g. email, failed

 Select receivers to consume: [email]:
  [0] email
  [1] failed
```

Reply:
seems like i have to verify everything ai says :-( sorry